src/Resources/doc/how-to-create-a-plugin.md

```diff
@@ -0,0 +1,324 @@
+# How to create a single-file plugin
+
+The possibility of creating plugins was introduced in version [7.0][1] of Guzzle Bundle and several plugins have already been published and downloaded thousands of times.
+
+Let’s see how easy it is to create your own single-file plugin!
+
+### Middleware
+
+> If you are not familiar with `middleware` definition, then first read this great article provided by Guzzle team: 
+> [http://docs.guzzlephp.org/en/stable/handlers-and-middleware.html#middleware][6]
+
+The primary goal of plugins is to provide possibility of configuration of middlewares for each guzzle client individually.
+
+So, first we need a middleware. Suppose we have one that adds some "magic header" with "magic value" to all the requests:
+
+```php
+// src/GuzzleMiddleware/MagicHeaderMiddleware.php
+
+namespace App\GuzzleMiddleware;
+
+use Psr\Http\Message\RequestInterface;
+
+class MagicHeaderMiddleware
+{
+    private $magicHeaderValue;
+
+    /**
+     * @param $magicHeaderValue
+     */
+    public function __construct($magicHeaderValue)
+    {
+        $this->magicHeaderValue = $magicHeaderValue;
+    }
+
+    /**
+     * @param callable $handler
+     *
+     * @return \Closure
+     */
+    public function __invoke(callable $handler) : \Closure
+    {
+        return function (
+            RequestInterface $request,
+            array $options
+        ) use ($handler) {
+            $request = $request->withHeader('magic-header', $this->magicHeaderValue);
+
+            // Continue the handler chain.
+            return $handler($request, $options);
+        };
+    }
+}
+```
+
+### Client
+
+Also we need a guzzle client, that will be used to perform requests:
+
+```yaml
+# config/packages/eight_points_guzzle.yaml
+
+eight_points_guzzle:
+    clients:
+        httpbin_client:
+            base_url: "http://httpbin.org"
+
+            plugin: ~
+```
+
+Now it’s time to connect middleware with the client through plugin system!
+
+The scope is to make working next configuration:
```

Comment:
maybe this sentence can be removed? seems to be unnecessary.

Reply:
yep, seems to be useless. Removed :slightly_smiling_face: